sample-plugin/src/main/java/com/github/jackchen/plugin/sample/instrumentation/SampleAsmClassVisitorFactory.kt

```diff
@@ -3,24 +3,52 @@ package com.github.jackchen.plugin.sample.instrumentation
 import com.android.build.api.instrumentation.AsmClassVisitorFactory
 import com.android.build.api.instrumentation.ClassContext
 import com.android.build.api.instrumentation.ClassData
-import com.android.build.api.instrumentation.InstrumentationContext
 import com.android.build.api.instrumentation.InstrumentationParameters
+import com.github.jackchen.android.sample.api.Extension
+import com.github.jackchen.android.sample.api.Register
+import com.github.jackchen.android.sample.api.TestCase
 import com.github.jackchen.plugin.sample.visitor.SampleClassVisitor
-import org.gradle.api.provider.Property
-import org.gradle.api.tasks.Input
-import org.gradle.api.tasks.Internal
-import org.gradle.api.tasks.Optional
 import org.objectweb.asm.ClassVisitor
-import java.io.File
 
+/**
+ * @author JackChen
+ */
+abstract class SampleAsmClassVisitorFactory :
+  AsmClassVisitorFactory<InstrumentationParameters.None> {
 
-abstract class SampleAsmClassVisitorFactory : AsmClassVisitorFactory<InstrumentationParameters.None> {
+  companion object {
+    private val ANNOTATION_CLASS_NAMES = hashSetOf<String>(
+      Register::class.java.name,
+      TestCase::class.java.name,
+      Extension::class.java.name,
+    )
+  }
 
-    override fun createClassVisitor(classContext: ClassContext, nextClassVisitor: ClassVisitor): ClassVisitor {
-        return SampleClassVisitor(nextClassVisitor)
-    }
+  override fun createClassVisitor(
+    classContext: ClassContext,
+    nextClassVisitor: ClassVisitor
+  ): ClassVisitor {
+    return SampleClassVisitor(nextClassVisitor)
+  }
 
-    override fun isInstrumentable(classData: ClassData): Boolean {
+  override fun isInstrumentable(classData: ClassData): Boolean {
+    val superClassName = classData.superClasses[0]
+    if (superClassName.isCompatActivityClassName() || superClassName.isFragmentActivityClassName()) {
```

Comment:
I think only using the superClass to determine the Activity or other components is inaccurate. 
We maybe have some enhanced classes that extend from AppcompatActivity.

Is it better to change the setContentView in Activity instead of changing the superclass directly?

Reply:
It feels a little better, but it's a lot more work and a lot more complicated than it is now, so we can think about it later.